cyhy_commander/nmap/nmap_importer.py

```diff
@@ -109,50 +112,96 @@ def __store_port_details(self, parsed_host):
-            details["time"] = time
+            details["time"] = parsed_host["endtime"]
             details["source"] = NmapImporter.SOURCE
-            details["owner"] = ip_owner
             details["latest"] = True
-            report = self.__db.PortScanDoc()
-            util.copy_attrs(details, report)
-            report.save()
-            if details.get("service", {}).get("name") in RISKY_SERVICES:
-                report["source_id"] = RISKY_SERVICES_SOURCE_ID
-                report["name"] = "Potentially Risky Service Detected: {}".format(
-                    details["service"]["name"]
-                )
-                report["service"] = details["service"]["name"]
-                self.__ticket_manager.open_ticket(
-                    report, "potentially risky service detected"
-                )
+
+            ticket_opened_for_owners = set()
+            if host_doc and host_doc.get("hostnames"):
+                # Create a PortScanDoc for each hostname/owner combination
+                for h in host_doc["hostnames"]:
+                    report = self.__db.PortScanDoc()
+                    util.copy_attrs(details, report)
+                    report["owner"] = h.get("owner", UNKNOWN_OWNER)
+                    report["hostname"] = h["hostname"]
+                    report.ip = ip  # sets ip and ip_int
+                    report.save()
+                    if details.get("service", {}).get("name") in RISKY_SERVICES:
+                        report["source_id"] = RISKY_SERVICES_SOURCE_ID
+                        report["name"] = "Potentially Risky Service Detected: {}".format(
+                            details["service"]["name"]
+                        )
+                        report["service"] = details["service"]["name"]
+                        # Open a ticket for this hostname/owner combination
+                        self.__ticket_manager.open_ticket(
+                            report, "potentially risky service detected"
+                        )
+                        ticket_opened_for_owners.add(h["owner"])
+            else:
+                # There are no hostnames in the HostDoc, so create a single
+                # PortScanDoc with no hostname that is owned by the IP owner
+                report = self.__db.PortScanDoc()
+                util.copy_attrs(details, report)
+                report["owner"] = ip_owner
+                report["hostname"] = None
+                report.ip = ip  # sets ip and ip_int
+                report.save()
+                if details.get("service", {}).get("name") in RISKY_SERVICES:
+                    # If IP owner is not the default owner and we didn't open a
+                    # ticket for them above, open a ticket for the IP owner
+                    if (
+                        ip_owner != DEFAULT_OWNER
+                        and ip_owner != UNKNOWN_OWNER
+                        and ip_owner not in ticket_opened_for_owners
```

Comment:
Given the `if`/`else` logic above can this ever be false? Either `host_doc and host_doc.get("hostnames")` is true and that branch of logic is evaluated, or it is false and this branch is evaluated. Since we set `ticket_opened_for_owners = set()` right before the `if` I don't see how we could get to here and have this statement evaluate to true.